### PR TITLE
Improve database ddl plan

### PR DESCRIPTION
**Pull request by harshit-gangal**
Closes: https://github.com/vitessio/vitess/issues/7035

### CODE REVIEW

File: go/vt/vtgate/planbuilder/builder.go [hunk 1]
```diff
@@ -353,6 +357,35 @@ func createInstructionFor(query string, stmt sqlparser.Statement, vschema Contex
 	return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "BUG: unexpected statement type: %T", stmt)
 }
 
+func buildDBDDLPlan(stmt sqlparser.Statement, vschema ContextVSchema) (engine.Primitive, error) {
```
Comment (systay) [inline go/vt/vtgate/planbuilder/builder.go:360]: `stmt` could be typed to `*sqlparser.DBDDL` in the function signature here

File: go/vt/vtgate/planbuilder/plan_test.go [hunk 2]
```diff
@@ -223,7 +223,7 @@ func TestWithDefaultKeyspaceFromFile(t *testing.T) {
 	// We are testing this separately so we can set a default keyspace
 	testOutputTempDir, err := ioutil.TempDir("", "plan_test")
 	require.NoError(t, err)
-	defer os.RemoveAll(testOutputTempDir)
+	//defer os.RemoveAll(testOutputTempDir)
```
Comment (systay) [inline go/vt/vtgate/planbuilder/plan_test.go:226]: intentional?

**Review by systay — APPROVED** [review]
A couple of nit picks, but LGTM

File: go/vt/vtgate/planbuilder/plan_test.go (hunk 2 above)
Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/plan_test.go:226]: no. :)

File: go/vt/vtgate/planbuilder/builder.go (hunk 1 above)
Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/builder.go:360]: this implements the function signature `f func(statement sqlparser.Statement, schema ContextVSchema)`

**Pull request merged**